pkg/adaptor/hypervisor/registry/register.go

```diff
@@ -6,5 +6,5 @@ import (
 )
 
 func NewServer(cfg hypervisor.Config, cloudConfig interface{}, workerNode podnetwork.WorkerNode, daemonPort string) hypervisor.Server {
-	return newServer(cfg, cloudConfig, workerNode, daemonPort)
+	return nil
```

Comment:
Can you please add this is a different commit? Also why is it returning nil?

Reply:
the `newServer` function, in fact is useless. 
as you can see ref to @huoqifeng 's comment above. https://github.com/confidential-containers/cloud-api-adaptor/pull/322#issuecomment-1290744174

the `NewServer` function is invoked base on build tag, and if we go through build tag this `newServer` function is not been invoked. or even if this file is able to be deleted? which I am not sure, hence I keep this file.

the reason I make this line to return `nil`
- `hypervisor.Server` is an interface, as the `NewServer` function seems not been invoked according to build tag is not set, I suppose the default value of an interface is `nil`
- as this file without build tag, during go vet process it will try to find `newServer` function as original code. But we don't have any `newServer` function implemented without any build tag. which breaks vet check.

Reply:
Thank you for the explaining. Makes sense

Reply:
@SamYuan1990 Looks like I'm missing something.
The `NewServer` method is the public method called from main. This invokes provider specific`newServer` based on build tag. If NewServer returns nil then how is it going to work?

Reply:
> @SamYuan1990 Looks like I'm missing something. The `NewServer` method is the public method called from main. This invokes provider specific`newServer` based on build tag. If NewServer returns nil then how is it going to work?

@bpradipt @SamYuan1990 I'm sorry I misunderstood it, right, it could not return nil. maybe, should we change registry to a factory and return concrete server in https://github.com/confidential-containers/cloud-api-adaptor/blob/staging/cmd/cloud-api-adaptor/main.go#L194? -- I got the same error report when build the repo in my pipeline system.

Reply:
@huoqifeng I would avoid changing it unless there is a good reason behind it.
Although to be honest, I don't understand this change :-( hence my question to @SamYuan1990 .
Btw, can you point me to the error that is reported ?

Reply:
wait a min, @huoqifeng , @bpradipt , if this function return nil breaks pipeline for build, why the build pipeline with tag successes?

Reply:
https://github.com/confidential-containers/cloud-api-adaptor/blob/staging/pkg/adaptor/hypervisor/ibmcloud/service.go#L55 em... will update this pr.

Reply:
@huoqifeng ， @bpradipt so far, I committed with a mock implementation which return nil when build tag is not given as an implementation of `newServer` to avoid error happen as go vet will not read build tag, as
```
# github.com/confidential-containers/cloud-api-adaptor/pkg/adaptor/hypervisor/registry
Error: pkg/adaptor/hypervisor/registry/register.go:9:9: undefined: newServer
make: *** [Makefile:59: vet] Error 2
```

Reply:
> @huoqifeng I would avoid changing it unless there is a good reason behind it. Although to be honest, I don't understand this change :-( hence my question to @SamYuan1990 . Btw, can you point me to the error that is reported ?

@bpradipt my company requires code scan before using it, the scan reported similar errors. Let me create an issue to address those issues reported separately.

Reply:
thanks @huoqifeng , I just updated code as adding build tag in go vet process. as you mentioned at https://github.com/confidential-containers/cloud-api-adaptor/issues/327